relay-metrics/src/aggregation.rs

```diff
@@ -1348,7 +1348,7 @@ impl Iterator for CappedBucketIter {
         }
 
         if current_batch.is_empty() {
-            self.next_bucket = None; // fuse
+            debug_assert!(self.next_bucket.is_none());
```

Comment:
Rather than a debug assert which we can never observe in production, let's please log here. However, we _also_ want to fuse the iterator correctly.